integrated_channels/integrated_channel/tasks.py

```diff
@@ -22,6 +25,34 @@
 User = auth.get_user_model()
 
 
+def locked(expiry_seconds, lock_name_kwargs):  # lint-amnesty, pylint: disable=missing-function-docstring
+    def task_decorator(func):
+        @wraps(func)
+        def wrapper(*args, **kwargs):  # lint-amnesty, pylint: disable=inconsistent-return-statements
+            cache_key = f'{func.__name__}'
+            for key in lock_name_kwargs:
+                cache_key = f'{cache_key}-{key}:{kwargs.get(key)}'
+            if cache.add(cache_key, "true", expiry_seconds):
+                exception = None
+                try:
+                    LOGGER.info('Locking task in cache with key: %s for %s seconds', cache_key, expiry_seconds)
+                    return func(*args, **kwargs)
+                except Exception as error:  # lint-amnesty, pylint: disable=broad-except
+                    LOGGER.exception(error)
+                    exception = error
+                finally:
+                    LOGGER.info('Unlocking task in cache with key: %s', cache_key)
+                    cache.delete(cache_key)
```

Comment:
I think this will execute even when the `try` block doesn't produce an exception, thus releasing the lock.  Should the `delete()` be moved to the `except` block?
https://github.com/openedx/edx-enterprise/pull/1589

Reply:
It depends on the problem, and the last time I thought about locking celery tests for an enterprise-catalog use case, I really wanted to lock the tasks _and_ not let the same `task(args)` run again within the expiry period.
It the goal is just to prevent the same `task(args)` from executing concurrently, then yeah, you can release the lock as soon as `task` is done.